lib/assert.js

```diff
@@ -119,37 +120,63 @@ assert.fail = fail;
 //                             expected: expected });
 assert.AssertionError = AssertionError;
 
-function getBuffer(fd, assertLine) {
+function getBuffer(fd, line, column) {
   let lines = 0;
   // Prevent blocking the event loop by limiting the maximum amount of
   // data that may be read.
   let maxReads = 64; // bytesPerRead * maxReads = 512 kb
   let bytesRead = 0;
-  let startBuffer = 0; // Start reading from that char on
-  const bytesPerRead = 8192;
-  const buffers = [];
-  do {
-    const buffer = Buffer.allocUnsafe(bytesPerRead);
+  let bytesPerRead = line === 0 ? column + 1000 : 8192;
```

Comment:
purely out of curiosity, why switch to reading 1000 bytes at a time rather than 8192? I'm guessing both values are somewhat arbitrary?

Reply:
It does not read 1000 bytes at a time but the exact number of bytes necessary. So we either safe reading to many bytes (the necessary bytes could be smaller than 8kb) or we safe extra read calls (we know that the necessary bytes are more than 8kb). So it just optimizes for these two cases and in all other cases it will still read exactly 8kb on each read.